### PR TITLE
Feature/filter events

### PR DESCRIPTION
**Pull request by danielcompton**
I just opened this PR so we can discuss it as you're working, and I can provide feedback early.

### CODE REVIEW

**Comment by danielcompton** [comment]
We can keep this branch as it is, but ideally in the future open one branch per feature so that it's easier to review and merge small changes.

File: src/day8/re_frame/trace.cljs [hunk 1]
```diff
@@ -113,80 +115,116 @@
...
+(defonce app-state [:traces {:filter-input ""
+                             :filter-items []}]) ;; [{:id (random-uuid) :query "showing" :filter-type "contains"} {:id (random-uuid) :query "Reagent" :filter-type "contains"}
+
+(defn render-trace-panel []
+  (let [filter-input     (r/atom "")
+        filter-items     (r/atom [])
         slower-than-ms   (r/atom "")
-        slower-than-bold (r/atom "")]
+        slower-than-bold (r/atom "")
+        filter-type      (r/atom "contains")]
```
Comment (danielcompton) [inline src/day8/re_frame/trace.cljs]: Clojure idioms usually have enums as keywords not strings, so `:contains`, not `"contains"`.

File: src/day8/re_frame/trace.cljs [hunk 2]
```diff
@@ -259,7 +305,26 @@
       (let [new-panel (.createElement js/document "div")]
         (.setAttribute new-panel "id" id)
         (.appendChild (.-body js/document) new-panel)
+        (js/window.focus new-panel)
         new-panel))))
 
+(defn inject-styles []
```
Comment (danielcompton) [inline src/day8/re_frame/trace.cljs]: I haven't looked too closely, but could https://google.github.io/closure-library/api/goog.style.html#installSafeStyleSheet be helpful here?

**Comment by daiyi** [comment]
Agreed that we went a little out of control here! We'll clean it up today and start with smaller branches!

**Comment by daiyi** [comment]
We've cleaned up the mega-branch and it's ready to be merged! 

re-frame-trace-filter-demo

Further work will be more sensibly contained within feature branches. \o/

File: src/day8/re_frame/trace.cljs [hunk 3]
```diff
@@ -195,19 +190,15 @@
            [:div.filter-control-input
             {:style {:margin-bottom 10}}
             [:select {:value @filter-type :on-change (fn [e]
-                                                         (reset! filter-type (.. e -target -value))
-                                                         (println (.. e -target -value)))}
-             [:option "contains"]
-             [:option "slower than"]]
+                                                         (reset! filter-type
+                                                                 (keyword (str/replace (.. e -target -value) " " "-"))))}
+             [:option :contains]
```
Comment (danielcompton) [inline src/day8/re_frame/trace.cljs]: Could you make the options like this, adding a value attribute?

```clj
     [:option {:value "option1"} "Option 1"]
     [:option {:value "option2"} "Option 2"]
     [:option {:value "option3"} "Option 3"]
     [:option {:value "option5"} "Option 5"]
```

Comment (daiyi) [inline src/day8/re_frame/trace.cljs]: done!

**Pull request merged**